pkg/config/generator_test.go

```diff
@@ -45,24 +45,28 @@ var primaryCNIConfig = map[string]interface{}{
 	"logfile-maxage":     5,
 }
 
-func newMultusConfigWithDelegates(pluginName string, cniVersion string, kubeconfig string, primaryCNIPluginConfig interface{}, configOptions ...Option) *MultusConf {
-	multusConfig := NewMultusConfig(pluginName, cniVersion, kubeconfig, configOptions...)
-	multusConfig.Delegates = []interface{}{primaryCNIPluginConfig}
-	return multusConfig
+func newMultusConfigWithDelegates(pluginName string, cniVersion string, kubeconfig string, primaryCNIPluginConfig interface{}, configOptions ...Option) (*MultusConf, error) {
+	multusConfig, err := NewMultusConfig(pluginName, cniVersion, kubeconfig, configOptions...)
+	if err != nil {
+		return multusConfig, err
+	}
+	err = multusConfig.Mutate(withDelegates(primaryCNIPluginConfig.(map[string]interface{})))
+	return multusConfig, err
 }
 
 func TestBasicMultusConfig(t *testing.T) {
-	multusConfig := newMultusConfigWithDelegates(
+	multusConfig, _ := newMultusConfigWithDelegates(
```

Comment:
Now that the method returns an error, it feels weird to swallow it up.

Do you mind asserting it does not error on the tests ? Or better; provide another helper for it ?